### DISCUSSION TITLE
I found that the remote connection speed on the web side is much faster than the client side speed

### DISCUSSION THREAD

**Discussion by youhers** [rustdesk/rustdesk#815] [General]
The client always needs to go around in circles for a long time to connect to the controlled end, but the web page does not. It can be connected with one click. What is the situation?

**Comment by rustdesk** [comment]
I guess the reason is the client need to attempt direct connection first (hole punch, NAT), but web not (always go relay).

**Comment by youhers** [comment]
I understand thank you

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/861
